modules/p2/ding_list/ding_list.module

```diff
@@ -0,0 +1,987 @@
+      DING_LIST_OPERATION_DETACH => TRUE,
+      DING_LIST_OPERATION_AUTO_CREATE => TRUE,
+      '_data' => array(
+        'default_title' => t('Loan history'),
+        'description' => t('Find your previous loans here. You can enable/disable the history in your profile.'),
+      ),
+    ),
+    DING_LIST_TYPE_BOOKS_READ => array(
+      DING_LIST_OPERATION_ATTACH => array('ting_object'),
+      DING_LIST_OPERATION_DETACH => TRUE,
+      DING_LIST_OPERATION_SORT => TRUE,
+      DING_LIST_OPERATION_AUTO_CREATE => TRUE,
+      '_data' => array(
+        'default_title' => t("Books I've read"),
+        'description' => t('Keep a collection of books you have read here.'),
+      ),
+    ),
+    DING_LIST_TYPE_REMEMBER => array(
+      DING_LIST_OPERATION_ATTACH => array('ting_object'),
+      DING_LIST_OPERATION_DETACH => TRUE,
+      DING_LIST_OPERATION_SORT => TRUE,
+      DING_LIST_OPERATION_AUTO_CREATE => TRUE,
+      '_data' => array(
+        'default_title' => t('Read later'),
+        'description' => t('Keep a collection of books you want to read later here.'),
+      ),
+    ),
+  );
+}
+
+/**
+ * Implements hook_ctools_plugin_directory().
+ */
+function ding_list_ctools_plugin_directory($owner, $plugin_type) {
+  return 'plugins/' . $plugin_type;
+}
+
+/**
+ * Implements hook_user_login().
+ *
+ * Synchronize with openlist, create missing lists and update the historical
+ * loans list.
+ */
+function ding_list_user_login(&$edit, $account) {
+  if (module_exists('ting_openlist')) {
+    ting_openlist_update_user($account);
+    
+    // Synchronize with openlist.
+    ding_list_sync_openlist($account);
+  }
+
+  // Create any missing auto created lists.
+  ding_list_setup_autocreated($account);
+
+  if (ding_user_consent_has_consent('loan_history_store')) {
+    // Update the historical loans.
+    ding_list_update_historical_loans($account);
+  }
+
+  ding_list_import_bookmarks($account);
+}
+
+/**
+ * Import from bookmarks.
+ */
+function ding_list_import_bookmarks($account) {
+  if (!module_exists('ding_bookmark')) {
+    return;
+  }
+  
+  // Using simplevar module here, because compatibility with lazy_vars is
+  // broken, will be fixed later.
+  if (simplevar_get('ding_list_import_bookmarks', $account->uid, FALSE) === TRUE) {
+    return;
+  }
+  
+  // Because we need the user credentials, the conversion is lazy, and not
+  // performed on install.
+  $bookmarks = ding_bookmark_load_bookmarks($account);    
+  if (empty($bookmarks)) {
+    simplevar_set('ding_list_import_bookmarks', $account->uid, TRUE);
+  }
+  
+  try {
+    $list = ding_list_get_list(DING_LIST_TYPE_REMEMBER);
+    
+    if ($list) {
+      foreach ($bookmarks as $bookmark) {
+        ding_list_create_ding_list_element($list, 'ting_object', array('value' => $bookmark->ding_entity_id), $account, FALSE);
+      }
+      
+      $list->save();
+      
+      simplevar_set('ding_list_import_bookmarks', $account->uid, TRUE);
```

Comment:
It seems like this will import from bookmarks on every login. Is that intentional?

I would expect a check for `ding_list_import_bookmarks` for `$account` first to avoid repeating the process.